src/Browser/Avalonia.Browser/Rendering/WebRenderTarget.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.JavaScript;
+using System.Threading;
+using Avalonia.Browser.Interop;
+using Avalonia.OpenGL;
+using Avalonia.Platform;
+using Avalonia.Reactive;
+using SkiaSharp;
+
+namespace Avalonia.Browser.Rendering;
+
+abstract partial class BrowserRenderTarget(JSObject js)
+{
+    protected readonly JSObject Js = js;
+
+    [JSImport("WebRenderTargetRegistry.getRenderTarget", AvaloniaModule.MainModuleName)]
+    private static partial JSObject? GetJsRenderTarget(int id);
+    
+    [JSImport("WebRenderTarget.setSize", AvaloniaModule.MainModuleName)]
+    private static partial void SetJsSize(JSObject target, int w, int h);
```

Comment:
I suspect having SetSize detached from the rendering/handling might make resizing performance worse. Especially with forced MT cross-thread marshalling.
But we can deal with that later.

Reply:
It's being called directly before starting a frame, so it's sort of auto-throttled by requestAnimationFrame